From b92e8ccd1db8936da0f829bd515b8e36d12fc049 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:23 -0400
Subject: [virt] kvm/ppc: Add signed type cast for comparation

Message-id: <1410545655-205645-35-git-send-email-dzickus@redhat.com>
Patchwork-id: 94082
O-Subject: [RHEL7 PATCH 034/626] powerpc/kvm: Add signed type cast for comparation
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 2fb10672c828f6e08f1dbe3690167300035adddc
Author: Chen Gang <gang.chen@asianux.com>
Date:   Mon Jul 22 14:32:35 2013 +0800

    powerpc/kvm: Add signed type cast for comparation

    'rmls' is 'unsigned long', lpcr_rmls() will return negative number when
    failure occurs, so it need a type cast for comparing.

    'lpid' is 'unsigned long', kvmppc_alloc_lpid() return negative number
    when failure occurs, so it need a type cast for comparing.

    Signed-off-by: Chen Gang <gang.chen@asianux.com>
    Acked-by: Paul Mackerras <paulus@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/book3s_hv.c b/arch/powerpc/kvm/book3s_hv.c
index 568c868..18cab29 100644
--- a/arch/powerpc/kvm/book3s_hv.c
+++ b/arch/powerpc/kvm/book3s_hv.c
@@ -1818,7 +1818,7 @@ static int kvmppc_hv_setup_htab_rma(struct kvm_vcpu *vcpu)
   rma_size <<= PAGE_SHIFT;
   rmls = lpcr_rmls(rma_size);
   err = -EINVAL;
-  if (rmls < 0) {
+  if ((long)rmls < 0) {
    pr_err("KVM: Can't use RMA of 0x%lx bytes\n", rma_size);
    goto out_srcu;
   }
@@ -1883,7 +1883,7 @@ int kvmppc_core_init_vm(struct kvm *kvm)
  /* Allocate the guest's logical partition ID */
 
  lpid = kvmppc_alloc_lpid();
- if (lpid < 0)
+ if ((long)lpid < 0)
   return -ENOMEM;
  kvm->arch.lpid = lpid;
 
-- 
1.7.1